Formula/mono.rb

```diff
@@ -38,16 +38,273 @@ class Mono < Formula
   link_overwrite "lib/cli"
 
   resource "fsharp" do
-    url "https://github.com/fsharp/fsharp.git",
-        tag:      "10.2.3",
-        revision: "e31bc96e8a5e5742af1c6c45d55d5cc06bb524cb"
+    url "https://github.com/dotnet/fsharp.git",
+        tag:      "v11.0.0-beta.20471.5",
+        revision: "03283e07f6bd5717797acb288cf6044cedca2202"
+    # F# patches hhen upgrading Mono, make sure to use the revision from
+    # https://github.com/mono/mono/blob/mono-#{version}/packaging/MacSDK/fsharp.py
+    patch do
+      url "https://raw.githubusercontent.com/mono/mono/a22ed3f094e18f1f82e1c6cead28d872d3c57e40/packaging/MacSDK/patches/fsharp-portable-pdb.patch"
+      sha256 "5b09b0c18b7815311680cc3ecd9bb30d92a307f3f2103a5b58b06bc3a0613ed4"
+    end
+    patch do
+      url "https://raw.githubusercontent.com/mono/mono/a22ed3f094e18f1f82e1c6cead28d872d3c57e40/packaging/MacSDK/patches/fsharp-netfx-multitarget.patch"
+      sha256 "112f885d4833effb442cf586492cdbd7401d6c2ba9d8078fe55e896cc82624d7"
+    end
+    patch do
+      url "https://github.com/dotnet/fsharp/commit/be6b22d11ae996b2d9b8e0724d9cf05ad65a0485.patch?full_index=1"
+      sha256 "793a39da798673b99289f3ac344ff8bd23d7eea2d3366c28e7e42229d8b130ca"
+    end
   end
 
   # When upgrading Mono, make sure to use the revision from
   # https://github.com/mono/mono/blob/mono-#{version}/packaging/MacSDK/msbuild.py
   resource "msbuild" do
     url "https://github.com/mono/msbuild.git",
-        revision: "ad9c9926a76e3db0d2b878a24d44446d73640d19"
+        revision: "db750f72af92181ec860b5150b40140583972c22"
+
+    # Remove in next release
+    # https://github.com/dotnet/msbuild/issues/6041
+    # https://github.com/dotnet/msbuild/pull/5381
+    patch do
+      url "https://github.com/mono/msbuild/commit/e2e4dfee543269ccb0a459263985b1c993feacec.patch?full_index=1"
+      sha256 "b64e93fbe1f5a5b8bcdb46ddd7d51a714f0e671b1b8ce2d1c2a0b80710ecb293"
+    end
+    patch do
+      url "https://github.com/mono/msbuild/commit/509c3190cf77be9422bddfad30b89e158f6229c3.patch?full_index=1"
+      sha256 "cf5fc342319cc1cb3b7bff02ec7d7d69af07f2777cf5b1910274d757cb14d92a"
+    end
+    patch do
+      url "https://github.com/mono/msbuild/commit/70bf6710473a2b6ffe363ea588f7b3ab87682a8d.patch?full_index=1"
+      sha256 "630b4187e882c162cd09e14f16ef2cca29b588dbea71bc444d925e5ef3f8f067"
+    end
+  end
+
+  def copy_fsharp_files_to_lib
```

Comment:
This is waaaayyyy too much custom install code on the homebrew side. All this should be patched upstream, or at least submitted upstream.

Reply:
@SMillerDev Upstream has similar code in their installation scripts: https://github.com/mono/mono/blob/mono-6.12.0.107/packaging/MacSDK/fsharp.py

Reply:
Why can't we use those scripts that are maintained by upstream then?

Reply:
@SMillerDev It script looks like connection with mono packaging infrastructure. It maybe not easy to integrate into homebrew build process. I just replicate their code on ruby it should not so hard to sync it in the future.

Reply:
It really needs to be maintained upstream.

Reply:
How many more files are there in `artifacts/bin/fsc/Release/net472` are there? Because if there's not too many, it might be more maintainable to just do
```ruby
(lib/"mono/fsharp").install Dir["artifacts/bin/fsc/Release/net*/*"]
```
Maybe there's a similar simplification for the other things you're doing here.

Reply:
I will try to contact maintainers to ask to extract F# packaing in a separate file. But I think it could be available only in future releases.

Reply:
> How many more files are there in `artifacts/bin/fsc/Release/net472` are there? Because if there's not too many, it might be more maintainable to just do
> 
> ```ruby
> (lib/"mono/fsharp").install Dir["artifacts/bin/fsc/Release/net*/*"]
> ```
> 
> Maybe there's a similar simplification for the other things you're doing here.

This folder has some additional files. An example localization folder.

Reply:
Is it ok for this version? Or I should do something with that?